src/lem/interpreter.rs

```diff
@@ -134,29 +166,80 @@ impl Block {
                     let a = bindings.get(a)?;
                     let b = bindings.get(b)?;
                     let c = match (a, b) {
-                        (Ptr::Leaf(Tag::Expr(Num), f), Ptr::Leaf(Tag::Expr(Num), g)) => {
+                        (Ptr::Leaf(_, f), Ptr::Leaf(_, g)) => {
+                            if g == &F::ZERO {
+                                bail!("Can't divide by zero")
+                            }
                             Ptr::Leaf(Tag::Expr(Num), *f * g.invert().unwrap())
                         }
                         _ => bail!("Division only works on numbers"),
                     };
                     bindings.insert(tgt.clone(), c);
                 }
+                Op::Lt(tgt, a, b) => {
+                    let a = bindings.get(a)?;
+                    let b = bindings.get(b)?;
+                    let c = match (a, b) {
+                        (Ptr::Leaf(_, f), Ptr::Leaf(_, g)) => {
+                            preimages.less_than.push(Some(PreimageData::FPair(*f, *g)));
+                            let f = Num::Scalar(*f);
+                            let g = Num::Scalar(*g);
+                            let b = if f < g { F::ONE } else { F::ZERO };
+                            Ptr::Leaf(Tag::Expr(Num), b)
+                        }
+                        _ => bail!("`<` only works on numbers"),
```

Comment:
Nit: here and in nearly every other top-level case, you don't need a match when capturing a single case. 
Possible alternatives:
```suggestion
                    let c = if let (Ptr::Leaf(_, f), Ptr::Leaf(_, g)) = (a, b) {
                        preimages.less_than.push(Some(PreimageData::FPair(*f, *g)));
                        let f = Num::Scalar(*f);
                        let g = Num::Scalar(*g);
                        let b = if f < g { F::ONE } else { F::ZERO };
                        Ptr::Leaf(Tag::Expr(Num), b)
                    } else {
                        bail!("`<` only works on numbers")
                     };
```
Or: https://github.com/huitseeker/match-opt